lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/agent/AgentData.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2025 Fraunhofer FOKUS and others. All rights reserved.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+
+package org.eclipse.mosaic.lib.objects.agent;
+
+import static org.apache.commons.lang3.builder.ToStringStyle.SHORT_PREFIX_STYLE;
+
+import org.eclipse.mosaic.lib.geo.GeoPoint;
+import org.eclipse.mosaic.lib.objects.UnitData;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+/**
+ * Holds relevant and volatile data about an agent, such as its
+ * position, state, or current transportation leg.
+ */
+public class AgentData extends UnitData {
+
+    private static final long serialVersionUID = 1L;
+
+    public enum State {
```

Comment:
After looking at the apps `State` seems a little bit too generic. I propose something along the lines of `CommuteState` or even `TransportationState` or maybe `<..>Status` is even better suitable.

Reply:
`TripStatus` seems suitable. 👍